### PR TITLE
Improve IPC to allow sending responses

### PR DESCRIPTION
**Pull request by smoogipoo**
1. Allows sending responses back to incoming requests.
2. Allows sending an IPC request with a response.
3. Extracts the port out from a const.

The goal here is to be able to accept IPC messages from osu!stable (e.g. to calculate difficulty) and send a response back.

### CODE REVIEW

File: osu.Framework/Platform/GameHost.cs [hunk 1]
```diff
@@ -111,7 +111,11 @@ public abstract class GameHost : IIpcHost, IDisposable
         /// </summary>
         protected virtual bool LimitedMemoryEnvironment => false;
 
-        protected void OnMessageReceived(IpcMessage message) => MessageReceived?.Invoke(message);
+        protected IpcMessage OnMessageReceived(IpcMessage message)
+        {
+            MessageReceived?.Invoke(message);
+            return null;
```
Comment (peppy) [inline osu.Framework/Platform/GameHost.cs:117]: Is the null return intended here? Shouldn't the `MessageReceived` become a `Func` and return a potential response here?

Comment (smoogipoo) [reply osu.Framework/Platform/GameHost.cs:117]: Yeah, I didn't want to make that change yet because it ripples throughout everything, but will adjust.

**Comment by peppy** [comment]
The deeper you look at this class, the more jank it looks. Pretty bad..

**Comment by smoogipoo** [comment]
Agree, definitely in need of a refactor.

File: osu.Framework/Platform/GameHost.cs [hunk 2]
```diff
@@ -94,7 +94,7 @@ public abstract class GameHost : IIpcHost, IDisposable
         /// </summary>
         public event Func<Exception, bool> ExceptionThrown;
 
-        public event Action<IpcMessage> MessageReceived;
+        public event Func<IpcMessage, IpcMessage> MessageReceived;
```
Comment (peppy) [inline osu.Framework/Platform/GameHost.cs:97]: let's xmldoc this while we're here. with the new method signature it's a bit hard to parse now.

**Review by peppy — APPROVED** [review]
